docs/plat/rpi3.rst

```diff
@@ -0,0 +1,297 @@
+ARM Trusted Firmware for Raspberry Pi 3
+=======================================
+
+.. section-numbering::
+    :suffix: .
+
+.. contents::
+
+The Raspberry Pi 3 is an inexpensive single-board computer that contains four
+Cortex-A53 cores, which makes it possible to have a port of the Arm Trusted
+Firmware.
+
+IMPORTANT NOTE: This port isn't secure. All of the memory used is DRAM, which is
+available from both the Non-secure and Secure worlds. This port shouldn't be
+considered more than a prototype to play with and implement elements like PSCI
+to support the Linux kernel.
+
+Design
+------
+
+The SoC used by the Raspberry Pi 3 is the Broadcom BCM2837. It is a strange SoC
+with a VideoCore IV that acts as primary processor (and loads everything from
+the SD card) and is located between all Arm cores and the DRAM.
+
+This explains why it is possible to change the execution state (AArch64/AArch32)
+depending on a few files on the SD card. We only care about the cases in which
+the cores boot in AArch64 mode.
+
+The rules are simple:
+
+- If a file called ``kernel8.img`` is located on the ``boot`` partition of the
+  SD card, it will load it and execute in EL2 in AArch64. Basically, it executes
+  a `default AArch64 stub`_ at address ``0x0``.
+
+- If there is also a file called ``armstub8.bin``, it will load it at address
+  ``0x0`` (instead of the default one) and execute it in EL3 in AArch64. All the
+  cores are powered on at the same time and start at address ``0x0``.
+
+This is ideal for us, as we can save the default AArch32 kernel in
```

Comment:
Why an AArch32 kernel? Presumably both AArch64 and AArch32 can be supported.

Reply:
I've added a short paragraph at the start of the documentation explaining the references to the AArch32 kernel. Also, the instructions to have a AArch64 kernel aren't that complicated, and most of this documentation applies to both kernels, so I've modified a bit the whole document in order to explain how to use both.